File: alligator/src/main/java/me/aartikov/alligator/DialogShowingListener.java
package me.aartikov.alligator;

/**
 * Date: 09.05.2017
 * Time: 16:11
 *
 * @author Artur Artikov
 */

/**
 * Interface for listening of dialog showing.
 */
public interface DialogShowingListener {
	/**
	 * Called when a dialog was shown.
	 *
	 * @param screenClass class of a screen that represents a dialog.
	 */
	void onDialogShown(Class<? extends Screen> screenClass);
}

File: alligator/src/main/java/me/aartikov/alligator/listeners/DialogShowingListener.kt
package me.aartikov.alligator.listeners

import me.aartikov.alligator.Screen

/**
 * Interface for listening of dialog showing.
 */
interface DialogShowingListener {
    /**
     * Is called when a dialog was shown.
     *
     * @param screenClass class of a screen that represents a dialog.
     */
    fun onDialogShown(screenClass: Class<out Screen>)
}